nats.go

```diff
@@ -149,6 +149,7 @@ var (
 	ErrConsumerConfigRequired       = errors.New("nats: consumer configuration is required")
 	ErrStreamSnapshotConfigRequired = errors.New("nats: stream snapshot configuration is required")
 	ErrDeliverSubjectRequired       = errors.New("nats: deliver subject is required")
+	ErrDurableNameRequired          = errors.New("nats: durable name of consumer is required")
```

Comment:
If we know the name of an ephemeral can we bind to that? I am not sure how that would act today or not but we may not want to restrict to durables here per se.

Reply:
We cannot right now due to some filter subject checks, but it might look something like this.  First need to get the name and delivery subject of the ephemeral then use bind stream and bind only to attach to that consumer without looking up the stream.

```go
	sub, err := js.SubscribeSync("foo")
	if err != nil {
		t.Fatalf("Unexpected error: %v", err)
	}
	cinfo, err := sub.ConsumerInfo()
	if err != nil {
		t.Fatal(err)
	}
	sub, err = js.SubscribeSync(cinfo.Config.DeliverSubject, nats.Durable(cinfo.Name), nats.BindStream("foo"), nats.BindOnly())
```

Reply:
ConsumerInfo should work for both durables and ephemerals.

Reply:
Which checks?

Reply:
I meant these [filter subject checks](https://github.com/nats-io/nats.go/blob/master/js.go#L1039-L1041) but found that it would work using the `nats.Durable` name for the consumer name even though it is ephemeral since that helps making the lookup of the consumer. Added examples of using BindOnly with ephemeral consumers, all these should work atm:

```go
	// Create ephemeral consumer
	sub1, err := js.SubscribeSync("foo")
	if err != nil {
		t.Fatalf("Unexpected error: %v", err)
	}
	cinfo, err := sub1.ConsumerInfo()
	if err != nil {
		t.Fatal(err)
	}

	// Bind to ephemeral consumer by setting the name using nats.Durable
	sub2, err := js.SubscribeSync("foo", nats.Durable(cinfo.Name), nats.BindStream("foo"), nats.BindOnly())
	if err != nil {
		t.Fatalf("Unexpected error: %v", err)
	}
	sub3, err := nc.SubscribeSync(cinfo.Config.DeliverSubject)
	if err != nil {
		t.Fatalf("Unexpected error: %v", err)
	}
```

Reply:
Feels off though to use Durable when its an ephemeral, at least to me.

In the call above with BindOnly, will we always need to provide the stream and consumer name? If so why not just make them args to BindOnly?